File: verve patch/base/Applications/CSymphony_0_1/csSamples/CodeMethodInvokeExpression.cs
using System;
using System.CodeDom;
using System.CodeDom.Compiler;
using System.Collections;
using CSymphony.Lib;

namespace CSymphony.CSy
{
	public class CSyUnit : GraphUnit
	{
		public override void Build()
		{
			ns("MyNameSpace");

			declClass(TypePublic, "MyClasss");
			declMethod(MemPublic, "System.Void", "MyMethod");

			MethodRef selfDispose = new MethodRef (my(), "Dispose");

			enter (ex (selfDispose, new Literal(true)));

			enter(ex (new MethodRef(my(), "SomeMethod"), new Literal (false)));

		}
	}
}

File: verve patch/base/Applications/CSymphony_0_1/vbSamples/CodeMethodInvokeExpression.vb
Option Strict Off
Option Explicit On

Imports System
Imports System.CodeDom
Imports System.CodeDom.Compiler
Imports System.Collections
Imports CSymphony.Lib

Namespace CSymphony.CSy
	Public Class CSyUnit
		 Inherits GraphUnit
		Overrides Public Sub Build()
			ns("MyNameSpace")

			declClass(TypePublic, "MyClasss")
			declMethod(MemPublic, "System.Void", "MyMethod")

			Dim selfDispose As MethodRef = New MethodRef (my(), "Dispose")

			enter (ex (selfDispose, newLit(true)))

			enter(ex (New MethodRef(my(), "SomeMethod"), newLit(false)))
		End Sub
	End Class
End Namespace
			